pywbem/cim_operations.py

```diff
@@ -636,6 +658,26 @@ def __init__(self, url, creds=None, default_namespace=DEFAULT_NAMESPACE,
         self._last_operation_time = None
         self._last_server_response_time = None
 
+        # logging
+        self.enable_logging = enable_logging
+
+        if self.enable_logging:
+            if enable_logging == 'all':
+                log_limit = None
+            elif enable_logging == 'min':
+                log_limit = DEFAULT_MAX_LOG_ENTRY_SIZE
```

Comment:
So "min" is really a default size? Maybe we should name the value like that, then.

Generally, I don't think it is very useful to truncate the fully detailed log data by size. What would be more useful is to allow different detail levels, and the maximum may well be truncated to a limit that is defined as a config variable (in the pywbem config module).

Reply:
**DISCUSSION.**

It is probably useful to do either/all of the above as long as we don't make this too complex. I really do worry about the costs of hacking around in the XML one more time.  In effect that means that we will be parsing it twice.

The thing that would be logical for http would be headers-only.

Reply:
I really do worry about hacking up the xml to pull out particular components.  That means another complete parse of the xml or integrating the logger with the parser.